### PR TITLE
Add genesis config to Glutton pallet

### PR DESCRIPTION
**Pull request by NachoPal**
Adding genesis config to Glutton pallet to avoid extra call to initialise Glutton Parachains.

### CODE REVIEW

File: frame/glutton/src/lib.rs [hunk 1]
```diff
@@ -105,6 +105,37 @@ pub mod pallet {
...
+		pub storage: Perbill,
+		pub trash_data_count: u32,
+	}
+
+	impl Default for GenesisConfig {
+		fn default() -> Self {
+			Self {
+				compute: Default::default(),
+				storage: Default::default(),
+				trash_data_count: Default::default(),
```
Comment (joepetrowski) [inline frame/glutton/src/lib.rs:120]: 5000?
https://github.com/paritytech/substrate/blob/polkadot-v0.9.42/frame/glutton/src/lib.rs#L146

Comment (NachoPal) [reply frame/glutton/src/lib.rs:120]: That's again a statement based on Kusama/Polkadot max PoV. I would keep it as it is, but I can change it to `5000` if you prefer.

Comment (gilescope) [reply frame/glutton/src/lib.rs:120]: Default is nicer.

**Review by gilescope — APPROVED** [review]

File: frame/glutton/src/lib.rs [hunk 2]
```diff
@@ -105,6 +105,37 @@ pub mod pallet {
...
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			<Compute<T>>::put(self.compute);
+			<Storage<T>>::put(self.storage);
+
+			if self.trash_data_count <= 65_000 {
```
Comment (bkchr) [inline frame/glutton/src/lib.rs:131]: That value is random? And the user will be confused why there is no trash data?

Comment (NachoPal) [reply frame/glutton/src/lib.rs:131]: > And the user will be confused why there is no trash data?
The trash data can be initialised from one of its main calls

Comment (bkchr) [reply frame/glutton/src/lib.rs:131]: Okay, but then it should panic if the value is bigger than 65_000.

Comment (joepetrowski) [reply frame/glutton/src/lib.rs:131]: > And the user will be confused why there is no trash data?

Yeah there should be an `else` statement to handle this (or just ensure before that it's below the pallet's limit).

Comment (bkchr) [reply frame/glutton/src/lib.rs:131]: @ggwpez also brought up a good idea, we should `hash` the data before we put it into the `TrashData`. This should ensure that the data isn't compressable that much.

Comment (NachoPal) [reply frame/glutton/src/lib.rs:131]: right, I overlooked that

Comment (ggwpez) [reply frame/glutton/src/lib.rs:137]: Yes it is using TwoX now, since that is faster and I assume we dont need the cryptographic property of Blake.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate-macos
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2874268

File: frame/glutton/src/lib.rs [hunk 3]
```diff
@@ -173,28 +222,31 @@ pub mod pallet {
...
 			Self::deposit_event(Event::ComputationLimitSet { compute });
 			Ok(())
 		}
 
-		/// Set the `Storage` storage value that determines the PoV size usage
-		/// for each block.
+		/// Set how much of the remaining `proof_size` weight should be consumed by `on_idle`.
+		//
+		/// 100% means that all remaining `proof_size` will be consumed. The PoV metering will
+		/// likely be an
```
Comment (joepetrowski) [inline frame/glutton/src/lib.rs:242]: be a what?

Comment (ggwpez) [reply frame/glutton/src/lib.rs]: Fixed. It is an over-estimation and will therefore consume less than intended.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2875061

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-each-crate
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2875075

**Review by joepetrowski — APPROVED** [review]

**Comment by NachoPal** [comment]
bot merge

**Pull request merged**